### ISSUE TITLE
Yarn install deletes parts of installs after cache is cleaned

### ISSUE THREAD

**Issue by gwkunze** [yarnpkg/yarn#3314]
**Do you want to request a *feature* or report a *bug*?**
*Bug*

**What is the current behavior?**
With certain packages (I think those that use binaries compiled with node-gyp) a second `yarn install` seems to delete part of the package, breaking the module.

**If the current behavior is a bug, please provide the steps to reproduce.**
With a simple package json:
```
{
  "name": "example",
  "version": "1.0.0",
  "main": "index.js",
  "license": "MIT",
  "dependencies": {
    "node-sass": "^4.5.2"
  }
}
```

- Run `yarn install`
- The directory `./node_modules/node_sass/vendor` should exist and contain a binary `binding.node`
- Clean the cache with `yarn cache clean`
- Run `yarn install --production` (the other way around, `--production` in the first step and without here has the same result)
- The directory `./node_modules/node_sass/vendor` no longer exists and node_sass doesn't work

**What is the expected behavior?**
Deleting a cache directory should have no effect on the results of `yarn install` (except for longer runtimes). It should especially not delete essential files.
 
**Please mention your node.js, yarn and operating system version.**
Tried with NodeJS 7.5, 7.7 and 7.10 in Docker

**Comment by gwkunze** [comment]
I did a `find .` in the directory before and after the second `yarn install`, this is the result of diff-ing the output:
```
--- original	2017-05-04 09:54:01.000000000 +0000
+++ second	2017-05-04 09:54:31.000000000 +0000
@@ -3746,9 +3746,6 @@
 ./node_modules/node-sass/test/scripts/util/proxy.js
 ./node_modules/node-sass/test/spec.js
 ./node_modules/node-sass/test/useragent.js
-./node_modules/node-sass/vendor
-./node_modules/node-sass/vendor/linux-x64-51
-./node_modules/node-sass/vendor/linux-x64-51/binding.node
 ./node_modules/nopt
 ./node_modules/nopt/.npmignore
 ./node_modules/nopt/.travis.yml
```

**Comment by bestander** [comment]
This is fixed in 0.24.3 now.
